backend/secrets/provider.go

```diff
@@ -83,27 +88,35 @@ func (m *ProviderManager) SubSecrets(command string) ([]string, error) {
 	providers := m.Providers()
 	// short circuit the function if there are no secrets providers
 	if len(providers) == 0 {
-		return secretVars, nil
+		return secretVars, fmt.Errorf("no secrets providers defined")
+	}
+	if m.Getter == nil {
+		return []string{}, fmt.Errorf("secrets management is not supported")
 	}
 
-	// iterate through each argument in the command
-	args := strings.Split(command, " ")
-	for _, a := range args {
-		// if the arg starts with $, find the secret
-		if strings.HasPrefix(a, "$") {
-			// iterate through each secrets provider
-			for name, p := range providers {
-				// ask the provider to retrieve the secret
-				secretKey := strings.TrimLeft(a, "$")
-				secretValue, err := p.Get(secretKey)
-				if err != nil {
-					logger.WithField("provider", name).WithError(err).Error("unable to retrieve secrets from provider")
-					return []string{}, err
-				}
-				if secretValue != "" {
-					secretVars = append(secretVars, fmt.Sprintf("%s=%s", secretKey, secretValue))
-				}
-			}
+	// iterate through each secret in the config
+	for _, secret := range secrets {
+		// get the provider name and secret ID associated with the Sensu secret
+		providerName, secretID, err := m.Getter.Get(ctx, secret.Secret)
+		if err != nil {
+			logger.WithField("provider", providerName).WithError(err).Error("unable to retrieve secrets from provider")
+			return []string{}, err
+		}
+		provider := providers[providerName]
+		if provider == nil {
+			err = fmt.Errorf("provider does not exist")
+			logger.WithField("provider", providerName).WithError(err).Error("unable to retrieve secrets from provider")
+			return []string{}, err
+		}
+		// ask the provider to retrieve the secret
+		secretKey := secret.Name
+		secretValue, err := provider.Get(secretID)
+		if err != nil {
+			logger.WithField("provider", providerName).WithError(err).Error("unable to retrieve secrets from provider")
```

Comment:
Do you think we should log the `secretID`? Or it's already included in the `err` returned by `provider.Get`?

Reply:
Good idea.. added it to the log!